cloudinit/config/cc_ssh_import_id.py

```diff
@@ -49,6 +50,15 @@
 
 def handle(_name, cfg, cloud, log, args):
 
+    if is_key_in_nested_dict(cfg, "ssh_import_id") and not subp.which(
+        SSH_IMPORT_ID_BINARY
+    ):
```

Comment:
Now that we actually have this function, what do you think about using it to early-exit anyway if no ssh_import_id is provided? This would allow us to walk through all users in the config dict a second time and emitting the logs `User (ubuntu|X|Y|Z) is not configured for ssh_import_id`

So what I'm proposing is this:
```suggestion
    if not is_key_in_nested_dict(cfg, "ssh_import_id"):
        log.debug("Skipping module named ssh-import-id, no 'ssh_import_id' directives found.")
        return
    elif not subp.which(SSH_IMPORT_ID_BINARY):
```